src/adaptations/device-layer/EventLoggingManager.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+ *
+ *    Copyright (c) 2019 Google LLC.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      Implementation for the Weave Device Layer EventLoggingManager object.
+ *
+ */
+
+#include <Weave/DeviceLayer/internal/WeaveDeviceLayerInternal.h>
+#include <Weave/DeviceLayer/EventLoggingManager.h>
+
+using namespace nl::Weave::DeviceLayer;
+
+namespace nl {
+namespace Weave {
+namespace DeviceLayer {
+namespace Internal {
+
+// For each priority level initialize event buffers and event Id counters.
+uint64_t gCritEventBuffer[(WEAVE_DEVICE_CONFIG_EVENT_LOGGING_CRIT_BUFFER_SIZE + 7) / 8];
```

Comment:
I'd recommend making a macro out of the ((n + 7) / 8) rounding math being performed here and using it consistently across all of the declarations.

There may already be a [macro to leverage from nlutilities](https://github.com/nestlabs/nlutilities/blob/master/include/nlmacros.h).

Reply:
Implemented and added ROUNDUP macro to the MathUtils.h file.